### PR TITLE
Enhance select command

### LINKED ISSUE #61: Implement an enhanced select function
so that user can backwards navigate

### PR DESCRIPTION
**Pull request by franceneee**
just type `select /nnotebook ` / `select /nnotebook /ssection` etc, it should work. always list to double-check the names of sections/ page numbers. if you would like to go back into `notebook_shelf` mode (when you list, you'll be able to see all the available notebooks), the command is `select /all`. not entirely sure if this `select /all` command is even that useful since you can get back into `notebook_shelf` mode by typing `mode /n`... well ya haha tell me if yall need me to do anything else. xD 

image

### CODE REVIEW

**Comment by franceneee** [comment]
should fix #62

File: src/main/java/seedu/duke/userinterface/InputParser.java [hunk 1]
```diff
@@ -91,35 +104,48 @@ private static boolean correctTimeFormat(String by) {
...
     public void extractNotebookParams(String argument, AppState appState)
-            throws InvalidNotebookException, InvalidSectionException, InvalidPageException {
+            throws InvalidNotebookException, InvalidSectionException, InvalidPageException,
+            NotebookOutOfBoundsException {
         Notebook notebook;
         Section section = null;
         String notebookTitle = parseNotebookTitle(argument);
         NotebookShelf notebookShelf = appState.getCurrentBookShelf();
         int notebookIndex = notebookShelf.findNotebook(notebookTitle);
         if (notebookIndex == -1) {
```
Comment (yAOwzers) [inline src/main/java/seedu/duke/userinterface/InputParser.java:147]: We should try to remove these -1 as it's a little confusing without any context

**Review by yAOwzers — COMMENTED** [review]
Ok good job!

**Review by yAOwzers — APPROVED** [review]

File: src/main/java/seedu/duke/userinterface/InputParser.java (hunk 1 above)
Comment (franceneee) [reply src/main/java/seedu/duke/userinterface/InputParser.java:147]: What do you suggest changing it to? xD Or you mean I should just replace that line with sth like `if (notebookIndex < 0)` ?

Comment (yAOwzers) [reply src/main/java/seedu/duke/userinterface/InputParser.java:147]: Oh like what's the significance of -1? Or its supposed to perform like `if (notebookIndex < 0)` ?

Comment (franceneee) [reply src/main/java/seedu/duke/userinterface/InputParser.java:147]: ooooooh the -1 is thrown when the notebook/section/page cannot be found (when the user selects a notebook/section/page that doesn't exist)... so you don't like that the -1 is being thrown is it ?

**Review by Lusi711 — APPROVED** [review]

**Pull request merged**